Extend/Controller/OrderController.php

```diff
@@ -0,0 +1,281 @@
+<?php
+/**
+ * Shop System Plugins:
+ * - Terms of Use can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/_TERMS_OF_USE
+ * - License can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/LICENSE
+ */
+
+namespace Wirecard\Oxid\Extend\Controller;
+
+use \OxidEsales\Eshop\Core\Field;
+use \OxidEsales\Eshop\Core\Registry;
+use \OxidEsales\Eshop\Application\Model\Basket;
+use \OxidEsales\Eshop\Core\Exception\ArticleInputException;
+use \OxidEsales\Eshop\Core\Exception\NoArticleException;
+use \OxidEsales\Eshop\Core\Exception\OutOfStockException;
+use \OxidEsales\Eshop\Application\Model\User;
+
+use \Wirecard\PaymentSdk\Response\FailureResponse;
+use \Wirecard\PaymentSdk\Response\InteractionResponse;
+use \Wirecard\PaymentSdk\Response\Response;
+use \Wirecard\PaymentSdk\Entity\Status;
+
+use \Wirecard\Oxid\Extend\Model\Payment_Gateway;
+use \Wirecard\Oxid\Extend\Model\Order;
+use \Wirecard\Oxid\Extend\Model\Payment;
+
+use \Psr\Log\LoggerInterface;
+
+/**
+ * Class Order
+ *
+ * @mixin \OxidEsales\Eshop\Application\Controller\OrderController
+ */
+class OrderController extends OrderController_parent
+{
+    /**
+     * Extends the parent init function and finalizes the order in case it was a Wirecard payment method
+     */
+    public function init()
+    {
+        parent::init();
+
+        $oConfig = Registry::getConfig();
+        $sWdPaymentRedirect = $oConfig->getRequestParameter('wdpayment');
+        $oSession = Registry::getSession();
+        $sWdSessionToken = $oSession->getVariable('wdtoken');
+
+        if ($this->_isPaymentFinished($sWdSessionToken, $sWdPaymentRedirect)) {
+            $sShopBaseUrl = $oConfig->getShopUrl();
+            $sLanguageCode = Registry::getLang()->getBaseLanguage();
+
+            $aParams = array(
+                'lang' => $sLanguageCode,
+                'force_sid' => $oSession->getId(),
+                'stoken' => $oSession->getSessionChallengeToken(),
+                'actcontrol' => 'order',
+                'cl' => 'order',
+                'fnc' => 'execute',
+                'sDeliveryAddressMD5' => $this->getDeliveryAddressMD5(),
+                'challenge' => '',
+                'ord_agb' => '1',
+                'oxdownloadableproductsagreement' => '0',
+                'oxserviceproductsagreement' => '0',
+                'wdtoken' => $sWdSessionToken
+            );
+
+            $sParamStr = http_build_query($aParams);
+
+            $sNewUrl = $sShopBaseUrl . 'index.php?' . $sParamStr;
+            header('Location: '.$sNewUrl);
```

Comment:
Please use the oxid method for redirect oxRegistry::getUtils()->redirect($sUrl, false);